core/src/main/java/discord4j/core/object/entity/channel/AllowedMentions.java

```diff
@@ -0,0 +1,222 @@
+package discord4j.core.object.entity.channel;
+
+import discord4j.discordjson.json.AllowedMentionsData;
+import discord4j.discordjson.json.ImmutableAllowedMentionsData;
+import discord4j.discordjson.possible.Possible;
+import discord4j.rest.util.Snowflake;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * A class for holding the allowed_mentions object with an built-in factory for default values.
+ * Also this class wraps the {@link AllowedMentionsData} JSON to a Discord4J class.
+ */
+public class AllowedMentions {
+
+    private static Possible<List<AllowedMentions.Type>> DEFAULT_TYPES = Possible.absent();
+    private static Possible<List<Snowflake>> DEFAULT_USERS = Possible.absent();
+    private static Possible<List<Snowflake>> DEFAULT_ROLES = Possible.absent();
```

Comment:
As convenient as it might be, I'm not comfortable with this global mutable state. Will it be common for users to want to set these defaults globally? I would've thought the allowed mentions would usually be tailored to each message. If it's still convenient, we should instead have `builder(Config)` and then a user can have their own global config if they'd like.

Reply:
I think this is less of a global mutual state, its more of a "default" in case nothing was provided to the builder.
For example a bot who doesn't want to ping everyone/here in any case, but always wants to parse roles could just do:
`AllowedMentions.setDefaultParsedTypes(Arrays.asList(AllowedMentions.Type.USER, AllowedMentions.TYPE.ROLE));`
And with calling Message#setAllowedMentions(AllowedMentions.builder().build()) he would be done.
If he wants to do some message specific changes, he can make them inside of the builder.

Yet I also understand your approach with the config. It probably makes more sense in some cases. I'll just go ahead and implement it that way.

Reply:
Another alternative would be to configure these defaults on the client. The downside with that is you then have to have something like `client.allowedMentionsBuilder()`.